gc/base/standard/ParallelScavengeTask.cpp

```diff
@@ -41,6 +41,14 @@ MM_ParallelScavengeTask::run(MM_EnvironmentBase *envBase)
 void
 MM_ParallelScavengeTask::setup(MM_EnvironmentBase *env)
 {
+	/* The alias threshold must be set before the scavenge tasks start but after the number of active threads
```

Comment:
The `Task::masterSetup(MM_EvnironmentBase *env)` call is likely the right place to put this functionality.

In `Task.hpp` see the following function description:
```
	/**
	 * Single call setup routine for tasks invoked by the master thread before the task is dispatched.
	 */
	virtual void masterSetup(MM_EnvironmentBase *env);
```

The `masterSetup` function is called in `Dispachter::run` before `prepareThreadsForTask` is called which means the number of threads that will be active in this GC bu after the override of the thread count set a few lines above. I think some slight refactoring of this code could clean things up and make sure the new code you are introducing is only being called once per GC

Reply:
There was actually intention to call this after prepareThreadsForTask which can change the number of actual threads used by the task.

There is no place after prepareThreadsForTask  that will ensure that any code will be executed by only one thread, but still before any of GC threads start doing work (without some extra tricks, like using atomics etc.)

Reply:
as for refactoring.... are you suggesting to extract recomputeActiveThreadCount (and few more lines around it) from prepareThreadsForTask  and call it just before masterSetup (still within Dispatcher::run() sequence)?

Reply:
Something like that... and it would be great if the recomputeActiveThreadCount used the value passed into the task setup that was added recently so there is only one place that thread count is changed.